palace/fem/multigrid.hpp

```diff
@@ -131,37 +131,27 @@ std::vector<std::unique_ptr<FECollection>> inline ConstructFECollections(
 }
 
 // Construct a hierarchy of finite element spaces given a sequence of meshes and
-// finite element collections. Dirichlet boundary conditions are additionally
-// marked.
+// finite element collections. Uses geometric multigrid and p multigrid.
 template <typename FECollection>
 inline mfem::ParFiniteElementSpaceHierarchy ConstructFiniteElementSpaceHierarchy(
     int mg_max_levels, bool mg_legacy_transfer, int pa_order_threshold,
     const std::vector<std::unique_ptr<mfem::ParMesh>> &mesh,
-    const std::vector<std::unique_ptr<FECollection>> &fecs,
-    const mfem::Array<int> *dbc_marker = nullptr,
-    std::vector<mfem::Array<int>> *dbc_tdof_lists = nullptr)
+    const std::vector<std::unique_ptr<FECollection>> &fecs, int dim = 1)
```

Comment:
Since we never use this to construct vector FE spaces, the `dim` argument can be removed. Further I'm against all of these changes as I've stated previously, since the previous code was fine. Optional arguments for the essential dofs are perfectly acceptable and I don't see how these changes improve things. All the changes to this file can be reverted for this PR.